apis/v1alpha2/util/translator/shared_types.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package translator
```

Comment:
are these functions meant to be used by implementors?

Reply:
Yes.

Reply:
I believe these and other httproute helpers were added by Contour maintainers, particularly to help with cases with pointer fields that have consts defined in the api, since you cant directly take a ref to a const

but of course as you say below, now we are on a go version with generics, this can all likely be refactored to use them

Reply:
most likely planning on getting rid of our usage of these with this: https://github.com/projectcontour/contour/pull/4905

so totally ok if these are removed

Reply:
Removed in #1585, amazing how much simpler this is with generics.